demo/src/main.rs

```diff
@@ -51,12 +51,15 @@ impl Drone<Hovering> {
 
 impl Drone<Flying> {
     fn has_arrived(&self, x: f32, y: f32) -> bool {
-        return self.x == x && self.y == y;
+        (self.x - x).abs() < f32::EPSILON && (self.y - y).abs() < f32::EPSILON
     }
 
     fn fly(mut self, x: f32, y: f32) -> Drone<Hovering> {
         self.x = x;
         self.y = y;
+        while !self.has_arrived(x, y) {
```

Comment:
I understand the motivation here, however without the increment this seems a bit pointless. Maybe I am missing something?

Reply:
I've added this function, so that the method `has_arrived()` is used, so that it does not throw a `unused function` warning. 

This solution was the simplest with being a somewhat realistic (assuming self.x and self.y are not set instantly, and it takes time to travel). I hadn't to delete any function or had to change the logic, so that the `Drone<Flying`> type state is exposed.

But as this is an example I guess it does not matter too much. :)

Reply:
In such case you could increment the variables, it becomes clearer to the readed. If you don't mind adding that change

Reply:
Ah I remember, why I have not made written it the obvious way. I produced an infinite loop, because of a logic error caused by myself, which I would've noticed, if I had paid attention :grinning:. 

Anyways, lets keep it simple and allow drowns to teleport :grin: